pkg/models/datasource_cache.go

```diff
@@ -139,7 +140,7 @@ func (ds *DataSource) HTTPClientOptions() (*sdkhttpclient.Options, error) {
 		}
 	}
 
-	if ds.JsonData != nil && ds.JsonData.Get("sigV4Auth").MustBool(false) {
+	if ds.JsonData != nil && ds.JsonData.Get("sigV4Auth").MustBool(false) && setting.SigV4AuthEnabled {
```

Comment:
Given almost all data sources been converted to use SDK contracts the usage of this should soon be gone.

Reply:
But we will still need to check for whether or not SigV4 config has been enabled, as that value along with the `jsonData` flag needs to be used in combination (due to original issue).

Alternatively, if there's some way of setting the `jsonData. sigV4Auth` to false when the `SigV4AuthEnabled ` config is disabled, or vice-versa

Reply:
Meant that we probably remove all of this code soon hopefully and rely on the HTTP client provider everywhere. 

Yeah probably good to have the check for now. You could do custom things thru provisioning for example.